src/Sarif.Multitool/FileWorkItemsOptions.cs

```diff
@@ -9,10 +10,40 @@ namespace Microsoft.CodeAnalysis.Sarif.Multitool
     public class FileWorkItemsOptions
     {
         [Option(
-            'f',
+            't',
             "filing-target-type",
             HelpText = "The type of system to which work items are to be filed. Must be one of 'github' and 'azureDevOps'.",
             Required = true)]
-        public string TargetType { get; internal set; }
+        public string FilingTargetType { get; internal set; }
+
+        [Option(
+            'p',
+            "project-uri",
+            HelpText = "The absolute URI of the project under which the work items are to be filed.",
+            Required = true)]
+        public string ProjectUriString { get; internal set; }
```

Comment:
>public string ProjectUriString { get; internal set; } [](start = 8, length = 53)

we have certain pieces of info being carried in "propertybag" of result e.g. the areapath. Should the Project uri also be embedded in there instead? What value are we bringing by having the user of the multitool pass this information. If the "area-path" doesnt exist in the result.PropertyBag, then we simply cannot log a bug. So, why split these two bound pieces of info being fed from two different places.

Reply:
This is a really good question. The basic reason, I think, is that a user might want to 'quarantine' issues in a test project by using a mirror. i.e., you might file bugs against your test project, which mirrors actual and then switch over. And so this has a practical use. Area paths have to be managed in some way, because there isn't necessarily a one-to-one mapping of a log file to an area path (i.e., you might want to group different results in different paths).

Having said the above, though, I wonder if we could get away with a single area path in our first scenario and then we'll side-step this weird intermediate data decoupling you've noticed. I think it's a good idea (ie, go ahead and add an area path arg and keep this out of SARIF for now. we will revisit when we double click into the more complex metadata universe). @lgolding?

---
In reply to: [306581368](https://github.com/microsoft/sarif-sdk/pull/1583#discussion_r306581368) [](ancestors = 306581368)

Reply:
1. If project URI is going to be in a property bag, it should be in `run.properties`, not `result.properties`. It's the same for all results in the run.
2. I did not understand the first paragraph of Michael's response. Let's mark this Pending and discuss further.

---
In reply to: [306584166](https://github.com/microsoft/sarif-sdk/pull/1583#discussion_r306584166) [](ancestors = 306584166,306581368)